packages/vsce/src/trees/CICSTree.ts

```diff
@@ -583,35 +557,21 @@ export class CICSTree implements TreeDataProvider<CICSSessionTree> {
 
   async updateSession(session: CICSSessionTree) {
     await ProfileManagement.profilesCacheRefresh();
-    const profileToUpdate = await ProfileManagement.getProfilesCache().loadNamedProfile(session.label?.toString()!, "cics");
-
-    const message = {
-      name: profileToUpdate.name,
-      profile: {
-        name: profileToUpdate.name,
-        ...profileToUpdate.profile,
-      },
-    };
-    await this.updateSessionHelper(session, message);
+    const profileCache = await ProfileManagement.getProfilesCache();
+    const profileToUpdate = profileCache.loadNamedProfile(session.label?.toString()!, "cics");
+    const currentProfile = await profileCache.getProfileFromConfig(profileToUpdate.name);
+    await this.updateSessionHelper(currentProfile);
   }
 
-  async updateSessionHelper(session: CICSSessionTree, messageToUpdate?: imperative.IUpdateProfile) {
-    const column = window.activeTextEditor ? window.activeTextEditor.viewColumn : undefined;
-    const panel: WebviewPanel = window.createWebviewPanel("zowe", `Update CICS Profile`, column || 1, { enableScripts: true });
-    panel.webview.html = addProfileHtml(messageToUpdate);
-    await panel.webview.onDidReceiveMessage(async (message) => {
-      try {
-        panel.dispose();
-        const profile = await ProfileManagement.updateProfile(message);
-        const position = this.loadedProfiles.indexOf(session);
-        await ProfileManagement.profilesCacheRefresh();
-        const updatedProfile = await ProfileManagement.getProfilesCache().loadNamedProfile(profile.profile.name, "cics");
-        await this.removeSession(session, updatedProfile, position);
-      } catch (error) {
-        // @ts-ignore
-        window.showErrorMessage(error);
-      }
+  async updateSessionHelper(profile?: imperative.IProfAttrs) {
+    const response = await window.showQuickPick([{ label: "Edit CICS Profile" }], {
+      ignoreFocusOut: true,
+      placeHolder: "Create a New Team Configuration File",
     });
+    if (response) {
+      const filePath = profile.profLoc.osLoc?.[0];
+      await openConfigFile(filePath);
+    }
```

Comment:
Sorry to be picky but, again, if osLoc is not set (which it doesn't have to be, it can be undefined), filePath can be set to undefined. Could do something similar to this, although you'd need to test and double check it, to only call openConfigFile with a valid input.

```suggestion
    if (response && profile.profLoc.osLoc?.[0]) {
      await openConfigFile(profile.profLoc.osLoc.[0]);
    } else {
      window.showErrorMessage("Some reasonable message about what you're trying to do??");
    }
```

Reply:
Actually, that condition you have added is redundant.
Update profile scenario will only be invoked on a available profile. So, If profile is available >> the Team Configuration file is available >> no need to check on` profile.profLoc.osLoc?.[0]`

Reply:
I have resolved this. We are falling back to Team Config File Path if `osLoc` fails to give us profile location path. This will cover us from all the angles.